providers/curation/github.js

```diff
@@ -131,6 +132,53 @@ class GitHubCurationService {
     return revision ? extend(true, {}, summarized, revision) : summarized;
   }
 
+  async getContent({ref, path}) {
+    const {owner, repo} = this.options;
+    const github = Github.getClient(this.options);
+    try {
+      const response = await github.repos.getContent({owner, repo, ref, path});
+      return base64.decode(response.data.content);
+    } catch (error) {
+      // @todo add logger
+    }
+  }
+
+  async postCommitStatus({
+    sha,
+    state,
+    description = 'ClealyDefined curation validation',
+    target_url = 'https://clearlydefined.io/'
+  }) {
+    const {owner, repo} = this.options;
+    const github = Github.getClient(this.options);
+    try {
+      await github.repos.createStatus({
+        owner,
+        repo,
+        sha,
+        state,
+        description,
+        target_url,
+        context: 'ClearlyDefined'
+      });
+    } catch (error) {
+      // @todo add logger
+    }
+  }
+
+  async getPrFiles({number}) {
+    const {owner, repo} = this.options;
+    const github = Github.getClient(this.options);
+    let files = [];
+    try {
+      const response = await github.pullRequests.getFiles({owner, repo, number});
+      files = response.data;
+    } catch (error) {
+      // @todo add logger
+    }
+    return files;
```

Comment:
we'd return files even in the event of an error?  Perhaps

```javascript
  try {
    return await github.pullRequests.getFiles({owner, repo, number}).data;
  } catch (error) {
    // @todo add logger
    throw error;
  }
```

Depending on the calling scenario, we may not even want to catch here. Rather catch and log up high.

Reply:
Sounds good.

Reply:
It didn't like returning .data directly from the async function. I had to first assign it to a var, then return `.data` after. But I made it throw the error 👍